### PR TITLE
Add Validation.getOrElseThrow(Function<E, X>)

### PR DESCRIPTION
**Pull request by sir4ur0n**
Closes https://github.com/vavr-io/vavr/issues/2693

### CODE REVIEW

File: src/test/java/io/vavr/control/ValidationTest.java [hunk 1]
```diff
@@ -300,6 +300,20 @@ public void shouldConvertFailureToU() {
...
+    public void shouldReturnValidWhenGetOrElseThrowWithFunctionOnValid() {
+      final Integer actual = Validation.<String, Integer> valid(1).getOrElseThrow(s -> new RuntimeException(s));
+      assertThat(actual).isEqualTo(1);
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void shouldThrowWhenGetOrElseThrowWithFunctionOnInvalid() {
+      Validation.<String, Integer> invalid("some error").getOrElseThrow(s -> new RuntimeException(s));
+    }
+
```
Comment (sleepytomcat) [inline src/test/java/io/vavr/control/ValidationTest.java:316]: We may also consider to add a test for an edge case, where `null` provided instead of `exceptionFunction`.

Comment (sleepytomcat) [reply src/test/java/io/vavr/control/ValidationTest.java:316]: Something like 
```
@Test(expected = NullPointerException.class)
public void shouldThrowNullPointerExceptionWhenExceptionFunctionIsNullOnValid() {
    Validation.<String, Integer> valid(1).getOrElseThrow(null);
}

@Test(expected = NullPointerException.class)
public void shouldThrowNullPointerExceptionWhenExceptionFunctionIsNullOnInvalid() {
    Validation.<String, Integer> invalid("some error").getOrElseThrow(null);
}
```

Comment (sir4ur0n) [reply src/test/java/io/vavr/control/ValidationTest.java:316]: Done (I adapted the test names a bit, and added types to make it compile)

Comment (sleepytomcat) [reply src/test/java/io/vavr/control/ValidationTest.java:316]: Thank you @sir4ur0n !
I believe we need @danieldietrich review and apporoval.

**Review by danieldietrich — APPROVED** [review]
Many thanks, well done!
